pkg/background/generate/generate.go

```diff
@@ -372,182 +372,215 @@ func (c *GenerateController) ApplyGeneratePolicy(log logr.Logger, policyContext
-		Namespace:  genNamespace,
-		Name:       genName,
-	}
-
-	genData, _, err := unstructured.NestedMap(genUnst.Object, "data")
-	if err != nil {
-		return noGenResource, fmt.Errorf("failed to read `data`: %v", err.Error())
-	}
-
-	genClone, _, err := unstructured.NestedMap(genUnst.Object, "clone")
-	if err != nil {
-		return noGenResource, fmt.Errorf("failed to read `clone`: %v", err.Error())
-	}
-
-	if len(genClone) != 0 {
-		rdata, mode, err = manageClone(logger, genAPIVersion, genKind, genNamespace, genName, policy.GetName(), genClone, client)
+	if rule.Generation.Clone.Name != "" {
+		cresp, mode, err = manageClone(logger, genAPIVersion, genKind, genNamespace, genName, policy.GetName(), rule.Generation, client)
+		rdatas = append(rdatas, GenerateResponse{
+			Data:          cresp,
+			Action:        mode,
+			GenName:       genName,
+			GenKind:       genKind,
+			GenNamespace:  genNamespace,
+			GenAPIVersion: genAPIVersion,
+			Error:         err,
+		})
+	} else if len(rule.Generation.CloneList.Kinds) != 0 {
+		rdatas = manageCloneList(logger, genNamespace, policy.GetName(), rule.Generation, client)
 	} else {
-		rdata, mode, err = manageData(logger, genAPIVersion, genKind, genNamespace, genName, genData, client)
-	}
-
-	if err != nil {
-		logger.Error(err, "failed to generate resource", "mode", mode)
-		return newGenResource, err
-	}
-
-	logger.V(3).Info("applying generate rule", "mode", mode)
-
-	if rdata == nil && mode == Update {
-		logger.V(4).Info("no changes required for generate target resource")
-		return newGenResource, nil
-	}
-
-	// build the resource template
-	newResource := &unstructured.Unstructured{}
-	newResource.SetUnstructuredContent(rdata)
-	newResource.SetName(genName)
-	newResource.SetNamespace(genNamespace)
-	if newResource.GetKind() == "" {
-		newResource.SetKind(genKind)
-	}
+		dresp, mode, err = manageData(logger, genAPIVersion, genKind, genNamespace, genName, rule.Generation, client)
+		rdatas = append(rdatas, GenerateResponse{
+			Data:          dresp,
+			Action:        mode,
+			GenName:       genName,
+			GenKind:       genKind,
+			GenNamespace:  genNamespace,
+			GenAPIVersion: genAPIVersion,
+			Error:         err,
+		})
+	}
+
+	logger.Info("applying the response data", "rdatas", rdatas)
+	for _, rdata := range rdatas {
+		if rdata.Error != nil {
+			logger.Error(err, "failed to generate resource", "mode", rdata.Action)
+			newGenResources = append(newGenResources, noGenResource)
+			return newGenResources, err
+		}
 
-	newResource.SetAPIVersion(genAPIVersion)
-	// manage labels
-	// - app.kubernetes.io/managed-by: kyverno
-	// "kyverno.io/generated-by-kind": kind (trigger resource)
-	// "kyverno.io/generated-by-namespace": namespace (trigger resource)
-	// "kyverno.io/generated-by-name": name (trigger resource)
-	common.ManageLabels(newResource, resource)
-	// Add Synchronize label
-	label := newResource.GetLabels()
+		logger.V(3).Info("applying generate rule", "mode", rdata.Action)
```

Comment:
It'd be good to continue on `rdata.Action==Skip`.

Reply:
Missed this?

Reply:
done